R/BASS_full_run.R

```diff
@@ -1,142 +1,122 @@
+#' @param n_samples The number of samples to draw in each sample
 #' @param return_all return each piece of BASS implementation
-#' @param seed_ random seed to use for random grts samples
-#' @param HexID_ Column for hexagon id
-#' @param stratumID Column for larger area id. Likely StudyAreaID or Province
-#' @param q Run the benefit calculation quickly
-#' @param non_ran_set Non random set that is added to the hypothetical sample set in benefit calculation.
-#' @param lakeN The land cover number to represent open water. 1 for FNLC, 18 for CLC
-#' @param benefit_weight The weight assigned to benefit in the selection probabilities.0.5 is equal weighting of cost and benefits. 1.0 is zero weighting to cost.
-#' @param noCost Do you only want to run the benefit calculation?
-#' @param weighted_benefits data frame with lc and weights
+#' @param seed random seed to use for random grts samples
+#' @param non_ran_set Non random set that is added to the hypothetical sample
+#'   set in benefit calculation.
+#' @param benefit_weight The weight assigned to benefit in the selection
+#'   probabilities.0.5 is equal weighting of cost and benefits. 1.0 is zero
+#'   weighting to cost.
+#' @param land_cover_weights data frame with lc and weights
+#'
+#' @inheritParams common_docs
 #'
 #' @return a table with inclusion probabilities
 #' @export
 #'
-full_BASS_run <- function(num_runs, nsamples, att, att.sp, cost, return_all = F, seed_ = as.integer(Sys.time()),
-                          HexID_ = HEX100, stratumID = StudyAreaID, q = F, non_ran_set = NULL, lakeN = 18,
-                          benefit_weight = 0.5, noCost = F, weighted_benefits = NULL) {
-  ## Check if habitat and sp object have same numbers of rows
-  if(nrow(att.sp)!= nrow(att)) rlang::abort(message = "att.sp and att must be the same size. They do not currently have the same number of rows")
-  ## Check if any NA in habitat columns
-  if(!any(grepl("^LC\\d",names(att)))) rlang::abort(message = "Habitat data table not formatted correctly. Try running clean_forBass.")
-  if (!"sf" %in% class(att)){
-    if(is.na(rowSums(summarise(att , across( matches("^LC\\d"), sum))))) rlang::abort(message = "Habitat data table not formatted correctly. Try running clean_forBass.")
-  }else{
-    if(is.na(rowSums(summarise(st_drop_geometry(att) , across( matches("^LC\\d"), sum))))) rlang::abort(message = "Habitat data table not formatted correctly. Try running clean_forBass.")
-  }
+#' @examples
+#'
+#' # With example data psu_hexagons and psu_costs...
+#'
+#' d <- full_BASS_run(
+#'   att_sf = psu_hexagons,
+#'   num_runs = 10,
+#'   n_samples = 3,
+#'   costs = psu_costs,
+#'   hex_id = hex_id)
+#'
+#' d <- full_BASS_run(
+#'   att_sf = psu_hexagons,
+#'   num_runs = 10,
+#'   n_samples = 0,
+#'   costs = psu_costs,
+#'   hex_id = hex_id)
 
-  set.seed(seed_)
+full_BASS_run <- function(att_sf, num_runs, n_samples, costs = NULL,
+                          hex_id, stratum_id = NULL, omit_flag = NULL,
```

Comment:
It's unclear to me how omit_flag is used in the function. It seems like it is called automatically in one of the check functions, but not sure.